### PR TITLE
PR #60: Integrating Cloud Translation API for translation capabilities

### PR DESCRIPTION
**Pull request by SriyaLingampalli**

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 1]
```diff
@@ -84,4 +100,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     response.sendRedirect("/user-page.html?user=" + recipient);
   }
+
+  // Translates text into target language
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: While it is not required for private functions, if you want to put a function level comment here, lets use Javadoc styling.

/** Translates text into target language. */

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 2]
```diff
@@ -59,10 +63,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
+      translateMessages(messages, targetLanguageCode);
+    }
+
     Gson gson = new Gson();
     String json = gson.toJson(messages);
 
     response.getWriter().println(json);
+
+    // Gets target language & calls helper function to translate
+    String targetLanguageCode = request.getParameter("language");
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: How does this work? Isn't this defined above already?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 3]
```diff
@@ -84,4 +100,23 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+
+      Translation translation =
+          translate.translate(originalText, TranslateOption.targetLanguage(targetLanguageCode));
+      String translatedText = translation.getTranslatedText();
+
+      message.setText(translatedText);
+    }
+  }
+
+  private void setText(String translatedText) {
```
Comment (FluffySheep) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Where do you use this method?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 2 above)
Comment (SriyaLingampalli) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Realized I included it twice. Removed in next commit

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 3 above)
Comment (SriyaLingampalli) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: In the process of implementing .setText at line 115

File: src/main/webapp/feed.html [hunk 4]
```diff
@@ -54,6 +54,11 @@
     return messageDiv;
   }
 
+  // Fetch data and populate the UI of the page.
+  function buildUI(){
```
[the remark this answers is not in the document]
Comment (FluffySheep) [reply src/main/webapp/feed.html]: Are you planning on putting this in a new pull request since it doesn't relate to cloud translation api?

**Comment by FluffySheep** [comment]
Please perform:

```
git reset --soft 7707cd18ffb1a06c3218aa71810979179fed690f
git rebase develop
```

**Pull request closed without merging**